aws/resource_aws_ecs_service_test.go

```diff
@@ -1955,3 +2027,41 @@ resource "aws_ecs_service" "test" {
 }
 `, rName, rName, rName, clusterName, tdName, svcName)
 }
+
+func testAccAWSEcsServiceWithSchedulingStrategy(clusterName, tdName, svcName string, schedulingStrategy *string, desiredCount *int) string {
+	schedulingStrategySnippet := ""
+	if schedulingStrategy != nil {
+		schedulingStrategySnippet += fmt.Sprintf(`scheduling_strategy = "%s"`, *schedulingStrategy)
+	}
+	desiredCountSnippet := ""
+	if desiredCount != nil {
+		desiredCountSnippet += fmt.Sprintf(`desired_count = %d`, *desiredCount)
+	}
+
+	return fmt.Sprintf(`
+resource "aws_ecs_cluster" "default" {
+  name = "%s"
+}
+resource "aws_ecs_task_definition" "ghost" {
+  family = "%s"
+  container_definitions = <<DEFINITION
+[
+  {
+    "cpu": 128,
+    "essential": true,
+    "image": "ghost:latest",
+    "memory": 128,
+    "name": "ghost"
+  }
+]
+DEFINITION
+}
+resource "aws_ecs_service" "ghost" {
+  name = "%s"
+  cluster = "${aws_ecs_cluster.default.id}"
+  task_definition = "${aws_ecs_task_definition.ghost.family}:${aws_ecs_task_definition.ghost.revision}"
+  %s
+  %s
+}
+`, clusterName, tdName, svcName, schedulingStrategySnippet, desiredCountSnippet)
+}
```

Comment:
I waver on this a bit and am personally a little inconsistent but I think I prefer having a bit of repetition in the test config code rather than more complicated dynamic config, particularly when it involves something as slightly convoluted as this.

More of a thought/comment than a critique but I think I find the tests a nice form of documentation about how to use things/how things work but this is a bit harder to follow here.

Reply:
I'll change it, as it seems other existing functions are also favoring repetition here.